dvc/remote/gdrive.py

```diff
@@ -324,10 +325,40 @@ def _create_remote_dir(self, parent_id, title):
 
     @gdrive_retry
     def _delete_remote_file(self, remote_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": remote_id}
         # it does not create a file on the remote
         item = self.drive.CreateFile(param)
-        item.Delete()
+
+        try:
+            item.Trash() if self._trash_only else item.Delete()
+        except ApiRequestError as exc:
+            http_error_code = exc.error.get("code", 0)
+            if (
+                http_error_code != 403
+                or self.list_params["corpora"] != "drive"
+            ):
+                raise
+            if (
+                len(exc.error.get("errors", []))
```

Comment:
make a small helper that extracts `location` - should be one simple line here

Reply:
Done in https://github.com/iterative/dvc/pull/3478/commits/9ebc1000211bfe0a83bb76b4b1ac5a8715c67efc